packages/atomic/src/components/common/generated-answer/render-answer-content.spec.ts

```diff
@@ -0,0 +1,311 @@
+      i18n,
+      // @ts-expect-error Test fixture with partial mock
+      generatedAnswerState: {
+        isStreaming: false,
+        answer: 'Test answer',
+        citations: [],
+        answerContentFormat: 'text/plain',
+        expanded: true,
+      },
+      isAnswerVisible: true,
+      hasRetryableError: false,
+      toggleTooltip: 'Toggle answer',
+      withToggle: false,
+      collapsible: false,
+      renderFeedbackAndCopyButtonsSlot: () => html``,
+      renderCitationsSlot: () => html``,
+      onToggle: vi.fn(),
+      onRetry: vi.fn(),
+      onClickShowButton: vi.fn(),
+      ...overrides,
+    };
+
+    const element = await renderFunctionFixture(
+      html`${renderAnswerContent({props: defaultProps})}`
+    );
+
+    return {
+      element,
+      props: defaultProps,
+    };
+  };
+
+  it('should call renderHeading with correct arguments', async () => {
+    await renderComponent();
+
+    expect(renderHeading).toHaveBeenCalledWith({
+      props: expect.objectContaining({
+        level: 0,
+        part: 'header-label',
+      }),
+    });
+  });
+
+  it('should call renderSwitch with correct arguments', async () => {
+    const onToggle = vi.fn();
+
+    await renderComponent({
+      isAnswerVisible: true,
+      withToggle: true,
+      toggleTooltip: 'Test tooltip',
+      onToggle,
+    });
+
+    expect(renderSwitch).toHaveBeenCalledWith({
+      props: expect.objectContaining({
+        part: 'toggle',
+        checked: true,
+        onToggle,
+        ariaLabel: 'Generated answer',
+        title: 'Test tooltip',
+        withToggle: true,
+        tabIndex: 0,
+      }),
+    });
+  });
+
+  describe('when hasRetryableError is true and answer is visible', () => {
+    it('should call renderRetryPrompt with correct arguments', async () => {
+      const onRetry = vi.fn();
+
+      await renderComponent({
+        hasRetryableError: true,
+        isAnswerVisible: true,
+        onRetry,
+      });
+
+      expect(renderRetryPrompt).toHaveBeenCalledWith({
+        props: expect.objectContaining({
+          onClick: onRetry,
+          buttonLabel: 'Retry',
+          message: expect.any(String),
+        }),
+      });
+    });
+
+    it('should not call renderRetryPrompt when answer is not visible', async () => {
+      await renderComponent({
+        hasRetryableError: true,
+        isAnswerVisible: false,
+      });
+
+      expect(renderRetryPrompt).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('when answer is visible and no retryable error', () => {
+    it('should call renderGeneratedContentContainer with correct arguments', async () => {
+      await renderComponent({
+        isAnswerVisible: true,
+        hasRetryableError: false,
+        // @ts-expect-error Test fixture with partial mock
+        generatedAnswerState: {
+          answer: 'Test answer',
+          answerContentFormat: 'text/markdown',
+          isStreaming: false,
```

Comment:
We should also test with isStreaming: true, answerContentFormat: 'text/plain' and another answer.